tests/ModelParser/JMSParserTest.php

```diff
@@ -166,6 +166,30 @@ public function providePropertyType(): iterable
             true,
             'string[]|null',
         ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("ArrayCollection<string>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'string[]|Collection|null',
+        ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("ArrayCollection<string, int>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'int[string]|Collection|null',
```

Comment:
the correct thing here would be `Collection<string>` i think.

the string representation is meant for debugging, we should have done `int<string>` i think, but we should not change that now to avoid a BC break.